### PR TITLE
Some updates and tweaks to Azure CI pipelines

### PR DESCRIPTION
**Pull request by MasterDuke17**
The main pipelines now use Ubuntu 20.04 (the latest available) and I added
single 18.04 and 16.04 pipelines for GCC and Clang (so four new in total)
so we still get some minimal testing on older versions.

### CODE REVIEW

**Review by patrickbkr — APPROVED** [review]
lgtm

File: azure-pipelines.yml [hunk 1]
```diff
@@ -243,3 +261,11 @@ stages:
             cp -v MoarVM-cover/merge-profraw.sh ../nqp/ &&
             ./html-cover.sh 2
           condition: and(succeeded(), eq( variables['COVERAGE'], 'yes' ))
+          displayName: Create coverage report
+
+        - script: |
+            valgrind --leak-check=full ../install/bin/raku --full-cleanup -e '' 2>valgrind.log
+            cat valgrind.log
+            ../install/bin/raku -ne 'BEGIN my @lost; @lost.push($0) if /"lost: " (\d+)/; END exit @lost.elems != 3 || all(@lost) != 0' valgrind.log
```
Comment (patrickbkr) [inline azure-pipelines.yml:269]: To make sure I understand this correctly: Three entries with each 0 lost bytes is expected?

File: azure-pipelines.yml [hunk 2]
```diff
@@ -49,55 +49,72 @@ stages:
...
            NQP_OPTIONS: '--backends=moar'
            MOAR_OPTIONS: '--has-libffi --cc=clang'
          MVM_coverage:
-           IMAGE_NAME: 'ubuntu-18.04'
+           IMAGE_NAME: 'ubuntu-20.04'
            NQP_OPTIONS: '--backends=moar'
            MOAR_OPTIONS: '--compiler=clang --coverage --optimize=0 --debug=3 --cc=clang'
-           COVERAGE: 1
-           USE_CLANG_LATEST: 1
+           COVERAGE: 'not working right now'
```
Comment (patrickbkr) [inline azure-pipelines.yml:99]: Given it's not working at the moment, can you disable the job entirely (commenting out?), to not needlessly burn up CPU cycles / speed up the run?

**Review by lizmat — APPROVED** [review]

**Review by patrickbkr — COMMENTED** [review]
Did we decide to just accept the occasional test failures on Windows caused by the `-j2` for now? (This is not meant as a suggestive question. I really am unsure if I've maybe missed a bit of discussion about this.)

File: azure-pipelines.yml (hunk 1 above)
Comment (MasterDuke17) [reply azure-pipelines.yml:269]: Yep, exactly correct. Looks like this:
```
==826695==    definitely lost: 0 bytes in 0 blocks
==826695==    indirectly lost: 0 bytes in 0 blocks
==826695==      possibly lost: 0 bytes in 0 blocks
```

Comment (patrickbkr) [reply azure-pipelines.yml:269]: OK. Thanks for the explanation!

**Comment by MasterDuke17** [comment]
> Did we decide to just accept the occasional test failures on Windows caused by the `-j2` for now? (This is not meant as a suggestive question. I really am unsure if I've maybe missed a bit of discussion about this.)

I don't think there was any definite yeas or nays. @ugexe said he prefers to only use a single job in his CI pipelines for reproducibility, but I don't believe he actually voted against this. I prefer using multiple jobs as that is more likely to uncover potential problems, even though it may make reproducing them harder. Since there are a couple approves for this PR I'll leave it as is (adding the `-j2`), but if there's a subsequent consensus against that's fine, we can revert.

In the multiple runs since that first one, I haven't seen that problem re-appear.

**Comment by patrickbkr** [comment]
Looking good. 👍🏽

File: azure-pipelines.yml (hunk 2 above)
Comment (MasterDuke17) [reply azure-pipelines.yml]: Done in 5fad749

**Pull request merged**